The OracleJDK Dehaunting

tomodachi94 opened the pull request:
"Unmaintained security disaster" -anonymous Nixpkgs maintainer

The ~~Ghostbusters~~ #jvm:nixos.org channel is coming to bust some ~~ghosts~~ old JDKs.

We dropped a few unsupported or unmaintained packages that happened to depend on OracleJDK.

Related: #313208

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

tomodachi94 committed:
javacard-devkit: drop

Had a dependency on the Oracle JDK,
as well as being several years out-of-date.

tomodachi94 committed:
docear: drop

Upstream itself says it is not actively developed:
https://docear.org/2017/10/05/update-for-the-google-scholar-parser-library-to-fetch-metadata-for-pdf-files/

The sole Nixpkgs maintainer has not used Docear in
4 years:
https://github.com/NixOS/nixpkgs/pull/353043#discussion_r1826528473

tomodachi94 committed:
oracle{jdk,jre}: drop

"Unmaintained security disaster" -anonymous Nixpkgs maintainer

emilazy commented:
@techknowlogick I see that you’ve made changes to the Atlassian packages recently – the NixOS module says that Atlassian only support the Oracle JDK, but reference a bug that appears to be related to some regex parsing issue of the `java -version` output? It seems like our OpenJDK packages should work fine with that, but also our Oracle JDK packages haven’t been updated since 2021(!!) and since they’re just builds of OpenJDK with a weird/bad licence, it seems like substituting our Nixpkgs‐built OpenJDKs should be fine. Just wanted to check that you think this should be fine, as I have no way of testing the modules.

techknowlogick commented:
@emilazy, thanks for the ping. I've since stopped using the Atlassian software suite, and I should probably rm my maintainership on those packages. TBH, I'd suggest making this switch as I'm not sure how much these are used with how few (see: none) reports came through for them being out of date, so if this PR breaks them, it might not have much impact.

emilazy commented on pkgs/applications/office/docear/default.nix:
```diff
@@ -25,7 +25,7 @@ stdenv.mkDerivation {
     # The wrapper ensures oraclejre is used
```
This comment is obsolete.

tomodachi94 replied:
Removed in latest push; it looks self-explanatory.

emilazy commented on pkgs/applications/office/docear/default.nix:
```diff
@@ -13,7 +13,7 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [ oraclejre ];
+  buildInputs = [ jre8 ];
```
https://github.com/NixOS/nixpkgs/pull/61817#discussion_r287624309 claims that this actually needs the Oracle JDK. Though if this is about `sun.*` packages, I *think* OpenJDK 8 would ship those? cc @unode – would you be able to test `docear` with this PR applied?

unode replied:
I could sometime next week but have to say that I haven't used docear in 4 years.

emilazy replied:
Thanks, no worries. I take it you are not interested in maintaining the package, then? As it has been unmaintained upstream since 2017 (and upstream [says as much](https://isg.beel.org/blog/2021/05/10/2021-update-for-docears-google-scholar-parser-library-to-fetch-metadata-for-pdf-files/)) and it doesn’t have a maintainer invested in it, perhaps we should simply drop this package instead.

unode replied:
I would be fine with dropping. We can always bring it back if someone wants it.

tomodachi94 replied:
Dropped as of the latest push, thank you for adding it in the first place ❤️

tomodachi94 committed:
sqldeveloper: drop

sqldeveloper was dropped due to being severely out-of-date and
having a dependency on JavaFX for Java 8, which we do not support.

emilazy commented on pkgs/top-level/all-packages.nix:
```diff
@@ -15710,9 +15685,7 @@ with pkgs;
-  sqldeveloper = callPackage ../development/tools/database/sqldeveloper {
-    jdk = oraclejdk;
-  };
+  sqldeveloper = callPackage ../development/tools/database/sqldeveloper { };
```
This presumably wants OpenJDK 8, and seemingly with JavaFX going by 294f667121f54f0ba5f77aedb38a0a9fa90bfa0f, which we do not support for OpenJDK 8, since AIUI it was not FOSS then; maybe Zulu or something might ship it? Also, following the `requireFile` instructions, I had to click on “Previous Version” 11 times to reach the download page for the corresponding version from January 2021. I think we ought to ditch this effectively‐unmaintained proprietary Oracle package too.

cc @ardumont who is listed as maintainer and @Ma27 who did the past several updates and who, in a commit message, said “I don't think that we should keep obviously unmaintained software that interacts with database systems.” :)

tomodachi94 replied:
I've gone ahead and simply dropped the package; I'm happy to undo that change if there is consensus to retain the package.

Ma27 replied:
Wow, past sins dug up I guess.

I had a use-case for that (Oracle DB for $customer and decent query plan integration for it), but that's not the case since ~2021, i.e. since I don't push updates for this. Given that nobody else seems to care... I'd say kill it with fire.

emilazy replied:
I have to admit I was shocked to see an upstanding citizen like you associated with an Oracle package 😆

Ma27 replied:
If you dig deep enough, there's a certain chance you'll find even more dirt (also tooling I needed for said project :see_no_evil:).

Anyways, I guess we can resolve this thread with sqldeveloper being removed.

tomodachi94 committed:
atlassian-{bamboo,confluence,crowd,jira}: drop

Support for the Atlassian Server products ended in
February 2024 and there was insufficient interest in
maintaining the Atlassian Data Center replacements

c.f. https://github.com/NixOS/nixpkgs/pull/353043#issuecomment-2452825826

tomodachi94 committed:
doc/languages-frameworks/java: remove reference to oraclejdk

About to be removed, and this doesn't look true anymore.

emilazy reviewed (approved):
## `nixpkgs-review` result                                                      
                                                                                
Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).    
                                                                                
Command: `nixpkgs-review pr 353043`                                             
                                                                                
---                                                                             
### `x86_64-linux`                                                              
<details>                                                                       
  <summary>:fast_forward: 1 package blacklisted:</summary>                      
  <ul>                                                                          
    <li>nixos-install-tools</li>                                                
  </ul>                                                                         
</details>

Where did all the packages go? 😕
